core/arch/riscv64/riscv64.asm

```diff
@@ -61,31 +61,29 @@ GLOBAL_LABEL(cpuid_supported:)
 GLOBAL_LABEL(call_switch_stack:)
         /* Init the stack. */
         addi     sp, sp, -32
```

Comment:
Should be `addi  sp, sp, -24` here?
```
________ Stack Bottom
|_s_ra_| 
|__s0__| 
|__s1__| _____SP
```

Reply:
No, `sp` needs to be aligned to 16bytes.